lib/SPIRV/PreprocessMetadata.cpp

```diff
@@ -207,8 +207,8 @@ void PreprocessMetadata::preprocessOCLMetadata(Module *M, SPIRVMDBuilder *B,
   // !{x} = !{i32 3, i32 102000}
   B->addNamedMD(kSPIRVMD::Source)
       .addOp()
-      .add(CLVer < kOCLVer::CL21 ? spv::SourceLanguageOpenCL_C
-                                 : spv::SourceLanguageOpenCL_CPP)
+      .add(CLVer == kOCLVer::CL21 ? spv::SourceLanguageOpenCL_CPP
```

Comment:
I am very confused about this statement, why CL version 2.1 corresponds to `SourceLanguageOpenCL_CPP`?

I think I am generally confused because the Translator seems to have more versions than what clang produces right now i.e. clang doesn't generate version 2.1 or 2.2.

In addition I don't know why we keep maintaining OpenCL C++. I think it only exists in a prototype which is on old version of LLVM and probably won't be compatible with IR on the newer releases. Would it be better if we just drop it completely?

Reply:
I agree with @AnastasiaStulova, I think we can make this simpler by just saying:

```suggestion
      .add(spv::SourceLanguageOpenCL_C)
```

and ignoring `spv::SourceLanguageOpenCL_CPP`.

Reply:
I'd be fine with this.  I wasn't sure who all is using "CL21"... especially since OpenCL C 2.1 doesn't exist!

Could we make this change as a follow-on PR?  IMHO we should remove the OCL21ToSPIRV passes also, so this would be part of the larger cleanup I described in the initial PR description.

Reply:
> Could we make this change as a follow-on PR?

That sounds reasonable to me, as your commit doesn't change this aspect.

Reply:
I agree, it is better to have clean up separately.

Reply:
FYI it was discussed on the call Oct 1st that `OCL21ToSPIRV` and `CL21`/`CL22` versions can likely be removed.

It seems however that `SourceLanguageOpenCL_CPP` is still used for SYCL. It is not clear whether this can be renamed, considering that there is no known frontend actively supporting OpenCL C++. I am looping in @bader for more feedback.

Reply:
Linking related discussion with @bader on https://github.com/intel/llvm/issues/2427

Reply:
I have a PR to remove `CL21` pass in #766